libraries/AP_DDS/AP_DDS_Client.cpp

```diff
@@ -586,18 +652,56 @@ bool AP_DDS_Client::create()
             requests[2] = dreader_req_id;
 
             if (!uxr_run_session_until_all_status(&session, requestTimeoutMs, requests, status, nRequests)) {
-                GCS_SEND_TEXT(MAV_SEVERITY_ERROR,"XRCE Client: Topic/Sub/Reader session request failure for index '%d'",(int)i);
+                GCS_SEND_TEXT(MAV_SEVERITY_ERROR,"XRCE Client: Topic/Sub/Reader session request failure for index '%zu'",i);
```

Comment:
what is %zu ?

Reply:
also see comments on debug code

Reply:
%zu is for size_t type. 

https://cplusplus.com/reference/cstdio/printf/

The `ARRAY_SIZE` macro calls `sizeof()` which returns a `size_t`. Since ArduPilot doesn't support containers with range-based for loops, we must manually set the type.